libsolidity/formal/SymbolicTypes.cpp

```diff
@@ -551,6 +551,11 @@ smtutil::Expression symbolicUnknownConstraints(smtutil::Expression _expr, fronte
 	solAssert(_type, "");
 	if (isEnum(*_type) || isInteger(*_type) || isAddress(*_type) || isFixedBytes(*_type))
 		return _expr >= minValue(_type) && _expr <= maxValue(_type);
+	else if (
+		auto arrayType = dynamic_cast<ArrayType const*>(_type);
+		arrayType && !arrayType->isDynamicallySized()
+	)
+		return smtutil::Expression::tuple_get(_expr, 1) == arrayType->length();
```

Comment:
Hm, wouldn't it be useful adding `&& (arrayType->length() >= 0)` (as below) here too?

Reply:
but that would just be redundant, here we're already assigning the exact value

Reply:
Ah `->length()` is the actual Solidity type, not a symbolic variable as the function name suggests.

Reply:
since `arrayType->length()` is not symbolic, but rather a concrete value at this point